lib/tasks/sprites.js

```diff
@@ -8,67 +8,120 @@ var notify = require("gulp-notify");
-        };
-
-        if (config.retinaSrcFilter || config.retinaImgName || config.retinaImgPath) {
-            spriteConfigs.retinaSrcFilter = [config.retinaSrcFilter];
-            spriteConfigs.retinaImgName = config.retinaImgName;
-            spriteConfigs.retinaImgPath = config.retinaImgPath;
+function createBitmapSprite(config, streams) {
+    var spriteConfigs = {
+        imgName: config.imgName,
+        imgPath: config.imgPath,
+        cssName: config.cssName,
+        cssTemplate: config.cssTemplate ? config.cssTemplate : __dirname + "/sprites.css_template.handlebars",
+        algorithm: "top-down",
+        algorithmOpts: {
+            sort: "sortImgsByHeight" in config ? config.sortImgsByHeight : true
         }
+    };
+
+    if (config.retinaSrcFilter || config.retinaImgName || config.retinaImgPath) {
+        spriteConfigs.retinaSrcFilter = [config.retinaSrcFilter];
+        spriteConfigs.retinaImgName = config.retinaImgName;
+        spriteConfigs.retinaImgPath = config.retinaImgPath;
+    }
+
+    var spriteData = gulp.src(config.imgSrc)
+        .pipe(spritesmith(spriteConfigs));
+
+    var imgStream;
+
+    if (config.imgOptim) {
+        imgStream = spriteData.img
+            .pipe(buffer())
+            .pipe(imagemin());
+    } else {
+        imgStream = spriteData.img;
+    }
+
+    imgStream.pipe(gulp.dest(config.destDir));
+
+    var outputLess = config.lessName !== undefined;
+
+    var cssStream = spriteData.css
+        .pipe(gulp.dest(config.destDir))
+        .pipe(gulpif(outputLess, rename(config.lessName)))
+        .pipe(gulp.dest(config.destDir));
 
-        var spriteData = gulp.src(config.imgSrc)
-            .pipe(spritesmith(spriteConfigs));
+    var stream = merge(imgStream, cssStream);
 
-        if (config.imgOptim) {
-            var imgStream = spriteData.img
-                .pipe(buffer())
-                .pipe(imagemin());
-        } else {
-            var imgStream = spriteData.img;
+    stream.pipe(notify({
+        onLast: true,
+        title: "Sprite created",
+        message: [
+            config.imgName,
+            config.retinaImgName,
+            config.cssName,
+            config.lessName
+        ].join(" "),
+    }));
+
+    streams.push(stream);
+}
+
+
+function createSvgSprite(config){
+    var spriteConfigs = {
+        spriteElementPath: config.imgSrc,
+        spritePath: config.imgName,
+        cssPath: config.cssName,
+        template: config.cssTemplate ? config.cssTemplate : __dirname + "/sprites-svg.css_template.handlebars",
+        svgo: {
+            plugins: [
+                { cleanupAttrs: true },
+                { cleanUpEnableBackground: true },
+                { cleanupIDs: true },
+                { minifyStyles: true }
+            ]
         }
+    };
 
-        imgStream.pipe(gulp.dest(config.destDir));
+    svgsprite(spriteConfigs, function(){
+        var pngFile = config.imgName.replace(/\.svg$/, ".png");
+        fs.unlinkSync(pngFile);
```

Comment:
This is the shameless part of this whole process... I looked for a node package that would generate svg sprites, work with gulp and would export css out of a handlebars template.

This `dr-svg-sprites` was it. But it turns out it also exports a PNG fallback that we are not really using and I don't think we need that. The worst part about that, is that there's no config option to turn it off, it will always export the PNG file.

So these 2 lines (88—89) are there only to get rid of the file 😅
It works, but I'm not particularly proud of it...